Tribler/Core/Modules/restapi/channels/channels_rss_endpoint.py

```diff
@@ -0,0 +1,184 @@
+
+
+class ChannelsRecheckFeedsEndpoint(BaseChannelsRssFeedsEndpoint):
+    """
+    This class is responsible for handling requests regarding refreshing rss feeds in your channel.
+    """
+
+    def render_POST(self, request):
+        """
+        .. http:post:: /channels/discovered/(string: channelid)/recheckfeeds
+
+        Rechecks all rss feeds in your channel. Returns error 404 if you channel does not exist.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X POST http://localhost:8085/channels/discovered/recheckrssfeeds
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "rechecked": True
+                }
+
+            :statuscode 404: if you have not created a channel.
+        """
+        channel_obj_dict = self.get_my_channel_obj_or_error(request)
+        if channel_obj_dict[0] is None:
+            return channel_obj_dict[1]
+
+        channel_obj_dict[0].refresh_all_feeds()
+
+        return json.dumps({"rechecked": True})
+
+
+class ChannelModifyRssFeedEndpoint(BaseChannelsRssFeedsEndpoint):
+    """
+    This class is responsible for methods that modify the list of RSS feed URLs (adding/removing feeds).
+    """
+
+    def __init__(self, session, cid, feed_url):
+        BaseChannelsRssFeedsEndpoint.__init__(self, session, cid)
+        self.feed_url = feed_url
+
+    def render_PUT(self, request):
+        """
+        .. http:put:: /channels/discovered/(string: channelid)/rssfeeds/http%3A%2F%2Ftest.com%2Frss.xml
+
+        Add a RSS feed to your channel. Returns error 409 if the supplied RSS feed already exists.
+        Note that the rss feed url should be URL-encoded.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X PUT http://localhost:8085/channels/discovered/abcd/rssfeeds/http%3A%2F%2Ftest.com%2Frss.xml
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "added": True
+                }
+
+            :statuscode 409: (conflict) if the specified RSS URL is already present in your feeds.
+        """
+        channel_obj_dict = self.get_my_channel_obj_or_error(request)
+        if channel_obj_dict[0] is None:
+            return channel_obj_dict[1]
+
+        channel_obj = channel_obj_dict[0]
+
+        if self.feed_url in channel_obj.get_rss_feed_url_list():
+            request.setResponseCode(http.CONFLICT)
+            return json.dumps({"error": "this rss feed already exists"})
+
+        channel_obj.create_rss_feed(self.feed_url)
+        return json.dumps({"added": True})
+
+    def render_DELETE(self, request):
+        """
+        .. http:delete:: /channels/discovered/(string: channelid)/rssfeeds/http%3A%2F%2Ftest.com%2Frss.xml
+
+        Delete a RSS feed from your channel. Returns error 404 if the RSS feed that is being removed does not exist.
```

Comment:
Yeah this style is also adopted in all other endpoints. The reason for this is that the documentation is meant for the developer and should explain what he should be doing with the endpoints.